src/Microsoft.DotNet.Build.Tasks/Targets/packageresolve.targets

```diff
@@ -23,9 +22,10 @@
     <ResolveNugetPackages>false</ResolveNugetPackages>
   </PropertyGroup>
 
-  <Target Name="RestorePackages" 
+  <!-- Restoring packages during a background (designtime) build will cause VS 2015 (v14) to get into an endless loop of resolving references. -->
+  <Target Name="RestorePackages"
           BeforeTargets="ResolveNuGetPackages"
-          Condition="'$(RestorePackages)'=='true'">
+          Condition="'$(RestorePackages)'=='true' and !('$(VSDesignTimeBuild)'=='true' and '$(VisualStudioVersion)' >= '14.0')">
```

Comment:
Unfortunately we can only do that if we do it in target as it isn't available until build time. I didn't want to introduce another dependent target for RestorePackages. I also wanted to keep the initial target I added central and clean as I expect we'll need to consume the property elsewhere.